pubsub/src/main/kotlin/com/clouway/pubsub/adapter/google/pubsub/JsonPubsubMessage.kt

```diff
@@ -1,11 +1,19 @@
 package com.clouway.pubsub.adapter.google.pubsub
 
+import com.google.appengine.repackaged.com.google.gson.Gson
+import java.time.LocalDateTime
 import java.util.*
 
 /**
  * @author Tsvetozar Bonev (tsbonev@gmail.com)
  */
 data class JsonPubsubMessage(val attributes: Map<String, String>, val data: String, val messageId: String){
+
+    fun getTime(): LocalDateTime {
+        val gson = Gson()
+        return gson.fromJson(attributes["time"], LocalDateTime::class.java)
```

Comment:
you can use type adapters like
```
class ISOLocalDateTimeAdapter : JsonSerializer<LocalDateTime>, JsonDeserializer<LocalDateTime> {

  override fun serialize(src: LocalDateTime?, type: Type, context: JsonSerializationContext): JsonElement? {
    val dateFormatAsString = DateTimeFormatter.ISO_LOCAL_DATE_TIME.format(src)
    return JsonPrimitive(dateFormatAsString)
  }

  override fun deserialize(json: JsonElement, type: Type, context: JsonDeserializationContext?): LocalDateTime? {
    return DateTimeFormatter.ISO_LOCAL_DATE_TIME.parse(json.asString, LocalDateTime::from)
  }
}
```
or you can transform the date to mills and then to parsi it as a date